server/controllers/finance/debtors/groups/index.js

```diff
@@ -230,24 +230,68 @@ function list(req, res, next) {
 /**
  * GET /debtor_groups/{:uuid}/invoices?balanced=1
  *
- * This function is responsible for getting all invoices of a specified debtor group
- *
  * @function invoices
- * @todo - implement this following the design in debtors/index.js
+ *
+ * @description This function is responsible for getting all invoices of a specified debtor group
+ *
  */
 function invoices(req, res, next) {
   var options = req.query;
 
-  if (options.balanced) { options.balanced = Boolean(options.balanced); }
+  loadInvoices(req.params.uuid)
+  .then(function (rows) {
+    let data = rows;
 
-  res.status(500).send('Unimplemented..');
+    if (options.balanced) {
+      data = rows.filter(item => {
+        return item.balance === 0;
+      });
+    }
 
-  /*
-  invoices(req.params.uuid, options)
-  .then(function (rows) {
-    res.status(200).json(rows);
+    if (options.unpaid) {
+      data = rows.filter(item => {
+        return item.balance !== 0;
+      });
+    }
+
+    res.status(200).json(data);
   })
   .catch(next)
   .done();
-  */
+}
+
+/**
+ * @method loadInvoices
+ * @description This method returns invoices of a debtor group
+ * @param {string} debtor_uuid A debtor group uuid
+ */
+function loadInvoices(debtor_uuid) {
+  // get debtors of the group
+  let sqlDebtors =
+    `SELECT uuid FROM debtor WHERE debtor.group_uuid = ?;`;
+
+  // get invoices balance for each debtor
+  let sqlInvoices =
+    `SELECT BUID(i.uuid) as uuid, i.date, CONCAT(project.abbr, i.reference) as reference,
+      debit, credit, (debit - credit) as balance, BUID(entity_uuid) as entity_uuid
+    FROM (
+      SELECT record_uuid as uuid, combined_ledger.date, SUM(debit) as debit, SUM(credit) as credit,
+        entity_uuid, invoice.reference, invoice.project_id
+      FROM combined_ledger
+      JOIN invoice ON combined_ledger.record_uuid = invoice.uuid OR combined_ledger.reference_uuid = invoice.uuid
+      WHERE entity_uuid IN (?) AND invoice.uuid NOT IN (SELECT voucher.reference_uuid FROM voucher WHERE voucher.type_id = 10)
```

Comment:
Instead of saying `voucher.type_id = 10`, could you give the "10" variable some context?  Like:

```js
const CASH_REVERSAL_TYPE = 10;

let sqlInvoices = `
  SELECT .....
   WHERE  .... AND invoice.uuid NOT IN (SELECT voucher.reference_uuid FROM voucher WHERE voucher.type_id = ${ CASH_REVERSAL_TYPE  })
  .... 
`;
```
I'm not sure, as a developer, what the magic number 10 means.